### ISSUE TITLE
Single call with `**kwargs` unpack emits many `invalid-argument-type` errors

### ISSUE THREAD

**Issue by stinodego** [astral-sh/ty#4160]
### Summary

This is not a bug (I think), but it is a usability concern.

When unpacking a dictionary as keyword arguments, an error is raised for every single argument of the callable that does not match the dictionary's value type. It is also impossible to tell from the error message which arguments are the problem.

You could argue the diagnostic is correct, but it is extremely noisy. My code base uses large Pydantic models, and used dict unpacking in some of our test suite. ty emitted hundreds of errors, which were easily fixed with only a handful of type fixes.

## Example

Playground: [link](https://play.ty.dev/f78655ae-edb0-413e-9354-7a42a67a9349)

```python
from dataclasses import dataclass

@dataclass
class Thing:
    a: int = 0
    b: bool = True
    c: bool = True
    d: bool = True
    e: bool = True

kwargs = {"a": 1}
t = Thing(**kwargs)  # raises 4 invalid-argument-type errors
```

ty output:

```
error[invalid-argument-type]: Argument is incorrect
  --> check.py:15:11
   |
15 | t = Thing(**kwargs)
   |           ^^^^^^^^ Expected `bool`, found `int`

error[invalid-argument-type]: Argument is incorrect
  --> check.py:15:11
   |
15 | t = Thing(**kwargs)
   |           ^^^^^^^^ Expected `bool`, found `int`

error[invalid-argument-type]: Argument is incorrect
  --> check.py:15:11
   |
15 | t = Thing(**kwargs)
   |           ^^^^^^^^ Expected `bool`, found `int`

error[invalid-argument-type]: Argument is incorrect
  --> check.py:15:11
   |
15 | t = Thing(**kwargs)
   |           ^^^^^^^^ Expected `bool`, found `int`
```

## Workaround

Currently, I am working around this issue by explicitly typing the kwargs dictionary as `dict[str, Any]`.

## Suggested improvement

In this case, I would expect a single error. Something along the lines of: "callable contains arguments not matching the unpacked dictionary".

### Version

ty 0.0.65
